src/froll.c

```diff
@@ -397,6 +397,277 @@ void frollsumExact(double *x, uint64_t nx, ans_t *ans, int k, double fill, bool
   }
 }
 
+/* fast rolling max */
+void frollmax(unsigned int algo, double *x, uint64_t nx, ans_t *ans, int k, int align, double fill, bool narm, int hasna, bool verbose) {
+  if (nx < k) {
+    if (verbose)
+      snprintf(end(ans->message[0]), 500, _("%s: window width longer than input vector, returning all NA vector\n"), __func__);
+    for (int i=0; i<nx; i++) {
+      ans->dbl_v[i] = fill;
+    }
+    return;
+  }
+  double tic = 0;
+  if (verbose)
+    tic = omp_get_wtime();
+  if (algo==0) {
```

Comment:
should `algo` be an `enum`, and we use `switch()` here?

It seems if we might add more `algo` options later, `enum` is the right storage type. o/w, if we only plan `0`/`1`, just do `if (algo == 0) { ... } else { ... }`

Reply:
I doubt we will have more than 2 options. This isn't really practical, unless some algo would be very memory efficient vs another time efficient, vs nested loop. Within rolling functions that are in the scope I doubt we will face such problem.

Reply:
`else if (algo==1)` has the advantage of being more verbose, so looking at the smaller chunk of code it is clear what is happening. Often github coversation tab will not show enough lines and one has to go to "Files" to see bigger picture. So that may be useful instead of a comment

Reply:
I think `if/else` keeps it simple then. `if (algo ==0) { ... } else { // algo==1` if we want to be so explicit.